### DISCUSSION TITLE
Multiple queues in a single node process

### DISCUSSION THREAD

**Discussion by daniel-white** [bbc/sqs-consumer#456] [Help] [answered]
### Summary

I know in other languages/runtimes with real threads, like .NET, I can create 2 separate queue pollers on separate threads. Node doesn't have a clean analog of that. 

If I were to design a service to consume more than 1 separate SQS queues, is it safe to have multiple `Consumer.create` instances, sharing an SQS client (or not)? Any indications on how long polling would function? Would it be "safe", meaning no race conditions between the consumers?

TIA.

**Comment by nicholasgriffintn** [comment, marked as answer]
Hey, sorry for the delay, this got lost in my notifications.

I believe this would be fine to do, however, I honestly don't know if we are doing this internally, generally, we are using this package for an individual queue not multiple.

I don't know if the community has any examples of where they are doing this, but I am sure I have heard from people who have before.

Sorry if that isn't very helpful.
